go/noise/gaussian_noise_test.go

```diff
@@ -445,19 +445,19 @@ func TestInverseCDFGaussian(t *testing.T) {
 	}{
 
 		{
-			desc:  "Abitrary input test",
+			desc:  "Arbitrary input test",
 			sigma: 1,
 			p:     0.05,
 			want:  -1.64485362695,
```

Comment:
Approx equal is only considers the 6 most significant digits. Having a want value of the form -1.64485362695 implies much greater accuracy than what is actually offered by the test. It may make sense to note the error tolerance of the test and prune the want values accordingly (unless you think the error tolerance will be adjusted later on)

Also, I was wondering how the want values are computed. They should not be computed by the code we are testing but via an independent method, e.g., Wolfram Alpha. When copmuting them make sure that they are truely exact up to the digits you are providing (Wolfram Alpha should be able to give you enough precision for that).

If necessary, please also fix for other Gaussian and Laplace tests.

On a side note, the sigma computed by the Gaussian noise is only accurate up to the 3 most significant digits. Thus, whenever your test something that implicitly uses a sigma computed by Gaussian noise, you can only test thefor the 3 most significant digits. To get an accurate want value, you should first compute a more accurate sigma (e.g. by ramping up the respective accuracy parameter of Gaussian noise) and then use Wolfram Alpha or something similar to get the respect bounds associated with this sigma.

Reply:
Yeah that is something I also wanted to change, perhaps a approxEqual function with Tolerance as a parameter is better.

For Laplace, The math is simple so I computed every step of the process using a high precision calculator (up to 22 digits). In laplace_noise_test_go, some of the cases has extra significant digits because I think the error tolerance should be adjusted accordingly for those values. I think the same case applies in gaussian_noise_test.go

For Gaussian for the 4 tests I added, I have used SigmaForGaussian function to compute the corresponding sigma and then calculate everything from there just as Laplace. I think this makes sense in terms we are testing the accuracy of the calculations for confidence interval not the accuracy of the sigma parameter. @laviniamnedelea can also provide her way of coming up with want values for Gaussian.

I can maybe change the gaussian test by computing a more accurate sigma and then recalculating all the want values, however I think the confidence interval will change significantly in some tests. Moreover, I think these tests should be only concerned with confidence interval calculations like I said previously.

The only problem I see with decrease the tolerance to something lower than 1e-6 (for ex 1e-10) is the inaccuracy that might result in Gaussian confidence interval computations since we use erfcInv function so I need to check how many digits are guaranteed to be accurate for gaussian.

Reply:
I wrote some additional tests for Java and it was accurate for the 3 most significant digits (but not more accurate, but the reason was most likely the sigma computation)

Happy to hear that you did independent validation. For the sigma cases, I think the right thing to do is compute a sigma that has the appropriate accuracy (unfrotunately, the only way I know how to do this easily is using the library with a bit more accuracy) and then use that accurate sigma in the manual coputation. Testing with 1e-3 tolerance should still accept. If not, we know there is a problem. (I can also send you some code from the Java library for reference (The context here is that we made computeQuantile part of the interface, so I added some extra tests for this)